app/src/main/java/com/nervousfish/nervousfish/activities/SharedActivityMethods.java

```diff
@@ -0,0 +1,143 @@
+                    return true;
+                }
+            }
+        } catch (final IOException e) {
+            LOGGER.error("Could not get contacts from database", e);
+        }
+        return false;
+    }
+
+    private static final class CreateNewContactClickListener implements SweetAlertDialog.OnSweetClickListener {
+        private final DialogInterface.OnClickListener confirmNewNameListener = new ConfirmNewNameClickListener();
+        private final Activity activity;
+        private final IDatabase database;
+        private final Contact contact;
+
+        CreateNewContactClickListener(final Activity activity, final IDatabase database, final Contact contact) {
+            this.activity = activity;
+            this.database = database;
+            this.contact = contact;
+        }
+
+        @Override
+        public void onClick(final SweetAlertDialog sweetAlertDialog) {
+            final AlertDialog.Builder alert = new AlertDialog.Builder(this.activity.getApplicationContext());
+            final EditText editText = new EditText(this.activity.getApplicationContext());
+
+            alert.setTitle(this.activity.getString(R.string.create_new_contact));
+            alert.setMessage(this.activity.getString(R.string.enter_name_contact));
+            alert.setView(editText);
+
+            alert.setNeutralButton(this.activity.getString(R.string.confirm), this.confirmNewNameListener);
+        }
+
+        // Suppressed because we only want the OnSweetClickListener to have access to this class
+        @SuppressWarnings({"InnerClassTooDeeplyNested", "NonStaticInnerClassInSecureContext"})
+        final class ConfirmNewNameClickListener implements DialogInterface.OnClickListener {
+            @Override
+            public void onClick(final DialogInterface dialog, final int which) {
+                LOGGER.info("Adding contact to database");
+                try {
+                    SharedActivityMethods.CreateNewContactClickListener.this.database.addContact(
+                            SharedActivityMethods.CreateNewContactClickListener.this.contact);
+                } catch (final IOException e) {
+                    LOGGER.error("Couldn't add the new contact", e);
+                }
+            }
+        }
+    }
+
+    private static final class AddPublicKeyToContactsClickListener implements SweetAlertDialog.OnSweetClickListener {
+        private final IDatabase database;
+        private final Contact contact;
+
+        AddPublicKeyToContactsClickListener(final IDatabase database, final Contact contact) {
+            this.database = database;
+            this.contact = contact;
+        }
+
+        @Override
+        public void onClick(final SweetAlertDialog sweetAlertDialog) {
+            try {
+                final Contact existingContact = this.database.getContactWithName(this.contact.getName());
+                existingContact.addKeys(this.contact.getKeys());
```

Comment:
But what happens when the existing contact also already has the public key? It looks like nothing happens